### PR TITLE
Routes links

### LINKED ISSUE #10: [1.0pt] Define Routes/links
- Ensure React-router is installed
- Configure Router by connecting components

### PR DESCRIPTION
**Pull request by HFG43**
Hi @AndreaM2429  Andre! I'm sending this pull request for you to review my code. Thanks in advance for your feedback!

Milestones 🎯:
📌 Separate Routes that required `authentication` from `login` & `register` pages
📌 Aplly conditional Render
📌 Generate `RoutesWrapper` component as child of route `/*` to ensure displaying all
📌 Ensure there is only one parent Router, and eliminate index Browser, as there should be only one in the app
📌 Generate paths for new components

🎖️ General requirements:
Gitflow
No linter errors
Ensure the use of best practices and Ruby style

### CODE REVIEW

File: src/Components/RoutesWrapper.js [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+  useEffect(() => {
+    if (status === 'succeeded') {
+      dispatch(getUserReservations(id));
+    }
+  });
+
+  if (userStatus === 'Authenticated') {
+    navigate('/login');
+  }
```
Comment (AndreaM2429) [inline src/Components/RoutesWrapper.js:33]: I can notice all your effort and good work in this feature, but when you navigate to the `MainPage` I get this error that is related to the multiple uses of `useSelector`.
image

To solve this problem consider supplying the following lines of code

```ruby
const userStatus = useSelector((state) => state.users.status);
const { id } = useSelector((state) => state.users.user);

# Suplay the two previous lines with the following one that will give you access to all the states of `users`
const userStore = useSelector((state) => state.users);
``` 

```ruby
dispatch(getUserReservations(id));
if (userStatus === 'Authenticated') 

# Look up for those lines and replace with
dispatch(getUserReservations(userStore.user.id));
if (userStore.status === 'Authenticated') 
```

File: src/Components/RoutesWrapper.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  return (
+    <Routes>
+      <Route path="MainPage" element={<MainPage />} />
```
Comment (AndreaM2429) [inline src/Components/RoutesWrapper.js:37]: Kindly fix this route, and add a `/` before `MainPage` like in the other routes.

File: src/App.js [hunk 3]
```diff
@@ -1,34 +1,37 @@
...
   return (
-    <div className="Routes" />
+    // <div className="Routes" />
+    <BrowserRouter>
+      <Routes>
+        <Route path="/login" element={<LoginPage />} />
+        <Route path="/" element={<Navigate to="/login" />} />
+        <Route path="/Register" element={<Register />} />
+        <Route path="/*" element={<RoutesWrapper />} />
+      </Routes>
```
Comment (AndreaM2429) [inline src/App.js:33]: Please check the redirection when the user is `Authenticated`, when I register the page redirects successfully to the `MainPage` but when I `Login` I can't access the `MainPage` and I continue in the `Login` page.

**Review by AndreaM2429 — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️ 

### Hi @HFG43  

I can see all your effort and good job trying to implement this new feature but there is some things you need to improve before merge.
Check the comments below.

## Highlights :1st_place_medal:

- Linters running.
- Good job applying the RoutesWrapper.
- Excellent idea to use a conditional render.

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval. Still, I would highly recommend that you consider these changes._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

----------------------------------------------------

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Comment by HFG43** [comment]
@AndreaM2429 Many thanks Andre for your feedback!! Please find the changes required applied!

**Review by AndreaM2429 — APPROVED** [review]
# Status: Approved :green_circle:

### Hi @HFG43 

You did a good work implementing the changes and making the app work as expected, your project is complete so it's time to merge it 💯 🥳.

congrats

## Highlights :1st_place_medal:

- Linters running
- The app redirects correctly when the user login
- The routes are working as expected

# Optional

_All the changes with optional tags are not crucial enough to prevent you from getting the approval. Still, I would highly recommend that you consider these changes._

Happy coding! :clap: :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**Pull request merged**